lsp/eslint.lua

```diff
@@ -0,0 +1,206 @@
+local util = require 'lspconfig.util'
+local lsp = vim.lsp
+
+-- @brief
+--
+-- https://github.com/hrsh7th/vscode-langservers-extracted
+--
+-- `vscode-eslint-language-server` is a linting engine for JavaScript / Typescript.
+-- It can be installed via `npm`:
+--
+-- ```sh
+-- npm i -g vscode-langservers-extracted
+-- ```
+--
+-- `vscode-eslint-language-server` provides an `EslintFixAll` command that can be used to format a document on save:
+-- ```lua
+-- vim.lsp.config('eslint', {
+--   --- ...
+--   on_attach = function(client, bufnr)
+--     vim.api.nvim_create_autocmd("BufWritePre", {
+--       buffer = bufnr,
+--       command = "EslintFixAll",
+--     })
+--   end,
+-- })
+-- ```
+--
+-- See [vscode-eslint](https://github.com/microsoft/vscode-eslint/blob/55871979d7af184bf09af491b6ea35ebd56822cf/server/src/eslintServer.ts#L216-L229) for configuration options.
+--
+-- Messages handled in lspconfig: `eslint/openDoc`, `eslint/confirmESLintExecution`, `eslint/probeFailed`, `eslint/noLibrary`
+--
+-- Additional messages you can handle: `eslint/noConfig`
+
+local function fix_all(opts)
+  opts = opts or {}
+
+  local bufnr = opts.bufnr or vim.api.nvim_get_current_buf()
+  vim.validate('bufnr', bufnr, 'number')
+
+  local client = opts.client or vim.lsp.get_clients({ bufnr = bufnr, name = 'eslint' })[1]
+
+  if not client then
+    return
+  end
```

Comment:
if the client doesn't exist why would the command be called? the command is defined on attach. and it's already passed by your call below, so get_clients isn't needed.

```suggestion
  local client = assert(opts.client)
```

Reply:
I think the intention was to just copy it without changing too much from the old config, but I agree.

This entire method could essentially just be inlined with 1-2 lines or so where the command is created